website/docs/getting-started/project-walkthrough.md

```diff
@@ -0,0 +1,169 @@
+  stack,
+  "SayFunction",
+  async (event: { message: string }) => {
+    // This code will run on AWS Lambda
+    console.log(event.message);
+    return;
+  }
+);
+```
+
+This bit of code offers us our first glimpse at the magic of Functionless.
+If you've used CDK before, you're probably familiar with the strict seperation it imposes between infrastructure code and runtime code.
+
+:::info
+
+Functionless lets you write runtime code inline with the code that defines the infrastructure for your runtime code.
+
+:::
+
+:::note
+
+**Runtime code** refers the application code you write that will run after the infrastructure defined by your `Stack` has been deployed.
+
+**Infrastructure code** refers to the CDK code used to configure the services that host your runtime code as well as any supporting services used by your runtime code.
+
+:::
+
+This opens up a world of new possibilities for how our runtime code can interface with our infrastructure.
+We can see this in action in the `StepFunction` Construct definition:
+
+```typescript
+new StepFunction(stack, "Workflow", async (event: { name: string }) => {
+  // This will run in an AWS Step Function
+  await sayFunction({ message: `Hello ${event.name}` });
+});
+```
+
+Functionless is doing three important things for us in this block of code:
+
+1. Allowing us to call the `sayFunction` Construct as a typed TypeScript function.
+2. Recognizing that we've called an Lambda Function and granting Step Functions permissions to invoke that Lambda Function.
+3. Converting a block of TypeScript code into Amazon States Language (ASL) for us.
+
+We're able to call the `sayFunction` Construct we just defined as if it were a standard TypeScript function call.
+Additionally, the arguments and return type of `sayFunction` match what we defined above:
+
+```typescript
+type sayFunction = (event: { message: string }) => Promise<void>;
+```
+
+:::info
+
+Functionless uses type-safety to ensure the shape of your data is consistent throughout your application.
+Changing your `Function` definition to accept a `number` instead of `string` for `message` would raise a TypeScript compiler error.
+Functionless maintains types across service boundaries eliminating data shape consistency errors and saving countless hours of toil.
+
+:::
+
+"But wait!", you might say. "Don't I have to set up IAM permissions for this Step Function to call that Lambda?"
+
+Fear not...
+
+:::info
+
+Functionless introspects how your runtime code interacts with its infrastructure and uses this information to automatically configure least-privilege policies.
+
+:::
+
+Finally, if you've ever created an AWS Step Function before, you might be wondering where all the ASL went.
+
+:::info
+
+Functionless automatically transpiles the code you supply in the `StepFunction` arrow function into ASL.
+This gives you the benefits of a world-class managed workflow service without requiring you to give up the productivity of TypeScript.
+
+:::
```

Comment:
I love how this is written but it was mentioned briefly above. Think it's OK to repeat?

Reply:
Yeah I think repetition is good for beginners. The admonitions also help more experienced users scan for just the important info and the repetition shouldn't be apparent to them.